pkg/nvml/mock/dgxa100/dgxa100.go

```diff
@@ -26,325 +26,49 @@ import (
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_1_SLICE,
-				SliceCount:            1,
-				InstanceCount:         4,
-				MultiprocessorCount:   1,
-				SharedCopyEngineCount: 4,
-				SharedDecoderCount:    2,
-				SharedEncoderCount:    2,
-				SharedJpegCount:       0,
-				SharedOfaCount:        0,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_2_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_2_SLICE,
-				SliceCount:            2,
-				InstanceCount:         2,
-				MultiprocessorCount:   2,
-				SharedCopyEngineCount: 4,
-				SharedDecoderCount:    2,
-				SharedEncoderCount:    2,
-				SharedJpegCount:       0,
-				SharedOfaCount:        0,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_4_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_4_SLICE,
-				SliceCount:            4,
-				InstanceCount:         1,
-				MultiprocessorCount:   4,
-				SharedCopyEngineCount: 4,
-				SharedDecoderCount:    2,
-				SharedEncoderCount:    2,
-				SharedJpegCount:       0,
-				SharedOfaCount:        0,
-			},
-		},
-		nvml.GPU_INSTANCE_PROFILE_7_SLICE: {
-			nvml.COMPUTE_INSTANCE_PROFILE_1_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_1_SLICE,
-				SliceCount:            1,
-				InstanceCount:         7,
-				MultiprocessorCount:   1,
-				SharedCopyEngineCount: 8,
-				SharedDecoderCount:    5,
-				SharedEncoderCount:    5,
-				SharedJpegCount:       1,
-				SharedOfaCount:        1,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_2_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_2_SLICE,
-				SliceCount:            2,
-				InstanceCount:         3,
-				MultiprocessorCount:   2,
-				SharedCopyEngineCount: 8,
-				SharedDecoderCount:    5,
-				SharedEncoderCount:    5,
-				SharedJpegCount:       1,
-				SharedOfaCount:        1,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_3_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_3_SLICE,
-				SliceCount:            3,
-				InstanceCount:         2,
-				MultiprocessorCount:   3,
-				SharedCopyEngineCount: 8,
-				SharedDecoderCount:    5,
-				SharedEncoderCount:    5,
-				SharedJpegCount:       1,
-				SharedOfaCount:        1,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_4_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_4_SLICE,
-				SliceCount:            4,
-				InstanceCount:         1,
-				MultiprocessorCount:   4,
-				SharedCopyEngineCount: 8,
-				SharedDecoderCount:    5,
-				SharedEncoderCount:    5,
-				SharedJpegCount:       1,
-				SharedOfaCount:        1,
-			},
-			nvml.COMPUTE_INSTANCE_PROFILE_7_SLICE: {
-				Id:                    nvml.COMPUTE_INSTANCE_PROFILE_7_SLICE,
-				SliceCount:            7,
-				InstanceCount:         1,
-				MultiprocessorCount:   7,
-				SharedCopyEngineCount: 8,
-				SharedDecoderCount:    5,
-				SharedEncoderCount:    5,
-				SharedJpegCount:       1,
-				SharedOfaCount:        1,
-			},
-		},
-	},
-}
-
 func New() nvml.Interface {
```

Comment:
Question: Does using the generated mocks add value here? 

It means that we don't have to implement an exhaustive interface and can rely on the stubbing by `moq` (or allow it to panic if an unimplemneted function is called).

Reply:
I'm not sure I follow. The types in this packag extend the generated mock type for exactly this purpose. I only overwrite the methods I care about, and the rest are left as stubs that panic.

Reply:
It actually how figured out which ones I needed to implement -- those that were missing were panicking via the moq stubs.

Reply:
One thing to note is that this breaks information that we get from the mock such as the call history -- number of calls and the parameters.

The "Correct" way to handle this is to implement the funcs as required. See for example: https://github.com/NVIDIA/go-gpuallocator/blob/82d8924b3d40c7d31cc18d56e65b3bffe59d91fd/gpuallocator/device_test.go#L28-L52

Reply:
Good point. Updated throughout.